packages/atomic/src/components/common/pager/pager-common.tsx

```diff
@@ -0,0 +1,136 @@
+import {Pager, SearchStatus} from '@coveo/headless';
+import {EventEmitter, h} from '@stencil/core';
+import ArrowRight from '../../../images/arrow-right.svg';
+import {FocusTargetController} from '../../../utils/accessibility-utils';
+import {randomID} from '../../../utils/utils';
+import {InsightPager, InsightSearchStatus} from '../../insight';
+import {InsightBindings} from '../../insight/atomic-insight-interface/atomic-insight-interface';
+import {Bindings} from '../../search/atomic-search-interface/atomic-search-interface';
+import {Button} from '../button';
+import {RadioButton} from '../radio-button';
+
+interface SearchStatusOptions {
+  bindings: Bindings | InsightBindings;
+  initializeSearchStatus(): SearchStatus | InsightSearchStatus;
+  initializePager(): Pager | InsightPager;
+  getEventEmitter(): EventEmitter | undefined;
+  getActivePage(): FocusTargetController | undefined;
+}
+
+export class PagerCommon {
+  private bindings: Bindings | InsightBindings;
+  private searchStatus: SearchStatus | InsightSearchStatus;
```

Comment:
We should not have any specific types for the `common` components. It will quickly spiral out of control otherwise. 

As a rule of thumb, it should only import types from the `common` folder.  

The way it could be done is that instead of having a `searchStatus` coming from Headless, you declare a type/interface with a prop `hasResult: boolean`.

Since that's the only thing we care about in this file, coming from the `searchStatus`.

And then in search/pager you map the specific `SearchStatus` to the "generic" `hasResults` prop. Same thing for insight panel pager.

Does it make sense ?

Reply:
ok, I actually did wonder about this @olamothe.. It does make sense, but I've done this for most components, so I might need to spend my last days redoing these types!